### PR TITLE
[K9VULN-4722] Fix Terraform Line Detection Errors

### PR DESCRIPTION
**Pull request by bahar-shah**
We were getting errors like `Failed to detect Terraform line, query response: %s` for a variety of reasons.
This PR attempts to address each of the edge cases that caused this.

* SearchKey wasn't properly sanitized in some cases so logic to match on the resource was erroring out
* Search keys with {} and [{{label}}] were failing to properly be matched
* Handle aws_something[resource_name] search keys
* Handle raw bracketed values (e.g., [abc]) vs placeholders (e.g., [{{var}}]) correctly
* Support Terraform resource matching within lists
* Handle replacement remediation case where it is an entire line that needs to be replaced and not just an in-line value
* For GCP, handle resource paths with "/" and not just "."
* Support multiple 'before' values separated by 'or' in remediation
* Handle nested blocks like metadata_options for remediations instead of erroring
* Handle quoted, unquotes, encoded string in remediation replacements

### CODE REVIEW

File: pkg/detector/helper.go [hunk 1]
```diff
@@ -83,19 +83,104 @@ func GetBracketValues(expr string, list [][]string, restOfString string) [][]str
 // '.' is new line
 // '=' is value in the same line
 // '[]' is in the same line
-func GenerateSubstrings(key string, extractedString [][]string) (substr1Res, substr2Res string) {
```
Comment (chouze) [inline pkg/detector/helper.go:86]: I'm struggling a bit to understand the purpose of this function, would you be able to add some simple input output examples?

File: pkg/detector/helper.go [hunk 2]
```diff
@@ -83,19 +83,104 @@ func GetBracketValues(expr string, list [][]string, restOfString string) [][]str
...
-	} else if parts := strings.Split(key, "="); len(parts) == valuePartsLength {
-		substr1, substr2 = getKeyWithCurlyBrackets(key, extractedString, parts)
-	} else {
-		parts := []string{key, ""}
-		substr1, substr2 = getKeyWithCurlyBrackets(key, extractedString, parts)
+
+	// Replace placeholders back to bracketed values
+	for idx, str := range extracted {
+		placeholder := fmt.Sprintf("{{%d}}", idx)
+		key = strings.Replace(key, placeholder, str[0], 1)
```
Comment (chouze) [inline pkg/detector/helper.go:92]: if we're only even pulling the first value from each sub slice, could this possibly be changed to a single dimensional slice?

File: pkg/report/model/sarif.go [hunk 3]
```diff
@@ -825,80 +839,169 @@ func TransformToSarifFix(vuln model.VulnerableFile, startLocation sarifResourceL
...
+				strings.HasPrefix(normalizedValue, altValue) ||
+				strings.HasPrefix(altValue, normalizedValue) {
+				matched = true
+				break
+			}
+		}
+
+		// Block-style fallback (e.g., nested blocks like metadata_options)
+		if !matched && strings.Contains(normalize(before), "{") && strings.Contains(normalize(before), "=") {
+			reInner := regexp.MustCompile(`(?m)(\w+)\s*=\s*(".*?"|\[.*?\]|\S+)`)
```
Comment (chouze) [inline pkg/report/model/sarif.go:896]: similar comment to extract the regex so it doesn't have to compile constantly

Comment (chouze) [reply pkg/report/model/sarif.go:896]: also a quick comment would be good for the regex as well

File: pkg/report/model/sarif.go [hunk 4]
```diff
@@ -825,80 +839,169 @@ func TransformToSarifFix(vuln model.VulnerableFile, startLocation sarifResourceL
 		before := strings.TrimSpace(patch["before"])
 		after := strings.TrimSpace(patch["after"])
 
-		// Regex to extract 'key = value' format
 		re := regexp.MustCompile(`(?m)["']?(?P<key>\w+)["']?\s*[:=]\s*(?P<value>\[.*?\]|".*?"|true|false|\d+)[,]?\s*`)
```
Comment (chouze) [inline pkg/report/model/sarif.go:842]: same about regex

**Review by chouze — APPROVED** [review]
nothing blocking

File: pkg/detector/helper.go (hunk 1 above)
Comment (bahar-shah) [reply pkg/detector/helper.go:86]: yeah- i'm planning on doing a final pass of comments/refactoring/tests once i've stabilized everything. Right now every edge case I find is causing lots of changes so waiting until I'm confident that its not changing much more

File: pkg/report/model/sarif.go (hunk 4 above)
Comment (bahar-shah) [reply pkg/report/model/sarif.go:842]: same- we can't compile statically because the keys are all different

File: pkg/detector/helper.go (hunk 2 above)
Comment (bahar-shah) [reply pkg/detector/helper.go:92]: ack- will do in a follow up
